platform/mbed_error.c

```diff
@@ -78,7 +78,9 @@ WEAK void error(const char *format, ...)
 
     //Call handle_error/print_error_report permanently setting error_in_progress flag
     handle_error(MBED_ERROR_UNKNOWN, 0, NULL, 0);
+#ifndef NDEBUG
     print_error_report(&last_error_ctx, "Fatal Run-time error");
```

Comment:
Shouldn't we guard everything in `error` function with NDEBUG? Should `handle_error` be called in case of NDEBUG?

Reply:
We should not guard everything. We still need handle_error to be called because we need to record if its first/last error which is always tracked and we will report it externally(using devicehealth apis for example) when required.

Reply:
But you could guard everything in `print_error_report()` with `NDEBUG` or have a dummy `print_error_report()` macro - neater than having multiple callers do ifdefs.

You should also watch out for "unused static function" warnings, which I think you'll get from the current version on some compilers.